app/src/main/java/com/android/unio/model/user/UserViewModel.kt

```diff
@@ -32,6 +32,10 @@ class UserViewModel(val repository: UserRepository, initializeWithAuthenticatedU
   }
 
   fun getUserByUid(uid: String, fetchReferences: Boolean = false) {
+    if (uid.isEmpty()) {
+      return
+    }
```

Comment:
Consider logging a warning when an empty uid is encountered. This can help with debugging and understanding user flow.